content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1107 @@
+
+Go to <https://docs.docker.com/engine/install/> to install the Docker Engine on your system.
+
+### [](https://github.com/interactivetech/e2e_blogposts/blob/main/ngc_blog/E2E-Part-1-Installation.md#2-download-the-tensorflow-container-from-the-ngc-catalog)2. Download the TensorFlow container from the NGC Catalog
+
+Once the Docker Engine is installed on your machine, visit <https://ngc.nvidia.com/catalog/containers> and search for the TensorFlow container. Click on the TensorFlow card and copy the pull command. [![](https://raw.githubusercontent.com/kbojo/images/master/NGC.png)](https://raw.githubusercontent.com/kbojo/images/master/NGC.png)
+
+Open the command line of your machine and past the pull command into your command line. Execute the command to download the container.
+
+`$ docker pull nvcr.io/nvidia/pytorch:21.11-py3`
+
+The container starts downloading to your computer. A container image consists of many layers; all of them need to be pulled.
+
+### [](https://github.com/interactivetech/e2e_blogposts/blob/main/ngc_blog/E2E-Part-1-Installation.md#3-run-the-tensorflow-container-image)3. Run the TensorFlow container image
+
+Once the container download is completed, run the following code in your command line to run and start the container:
+
+`$ docker run -it --gpus all -p 8888:8888 -v $PWD:/projects --network=host nvcr.io/nvidia/pytorch:21.11-py3` 
+
+[![](https://raw.githubusercontent.com/kbojo/images/master/commandline1.png)](https://raw.githubusercontent.com/kbojo/images/master/commandline1.png)
+
+### [](https://github.com/interactivetech/e2e_blogposts/blob/main/ngc_blog/E2E-Part-1-Installation.md#4-install-jupyter-lab-and-open-a-notebook)4. Install Jupyter lab and open a notebook
+
+Within the container, run the following commands:
+
+`pip install torchvision==0.11.3 jupyterlab`
+
+`jupyter lab --ip=0.0.0.0 --port=8888 --allow-root`
+
+Open up your favorite browser and enter: [http://localhost:8888/?token=*yourtoken](http://localhost:8888/?token=*yourtoken)*. [![](https://raw.githubusercontent.com/kbojo/images/master/commandline2.png)](https://raw.githubusercontent.com/kbojo/images/master/commandline2.png)
+
+You should see the Jupyter Lab application. Click on the plus icon to launch a new Python 3 notebook. Follow along with the image classification with the TensorFlow example provided in Part 2. 
+
+Now that we have our Docker Engine installed and the PyTorch Container running, we need to fetch and prepare our training dataset. That’s coming up in Part 2 below.
+
+- - -
+
+# Part 2: Data Preparation
+
+Note this Demo is based on ngc docker image `nvcr.io/nvidia/pytorch:21.11-py3`
+
+This notebook walks you each step to train a model using containers from the NGC Catalog. We chose the GPU optimized Pytorch container as an example. The basics of working with docker containers apply to all NGC containers.
```

Comment:
This is a duplicate sentence, identical to the sentence in Part 1, Installation section.
Do you need to repeat this information in the post, part 2?

Reply:
Yes, let's keep it.